src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+  const nameRegex = /^[A-Za-z\s]+$/;
+
+  if (formName.length < 4 || !nameRegex.test(formName)) {
+    pushNotification(
+      'Name must be at least 4 characters long and contain only letters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateAge(age) {
+  if (age < 18 || age > 90) {
+    pushNotification('Age must be between 18 and 90.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validatePosition(position) {
+  if (!position.trim()) {
+    pushNotification('Position is required.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateOffice(office) {
+  if (!office.trim()) {
+    pushNotification('Office is required.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateSalary(salary) {
+  const parsedSalary = parseFloat(salary);
+
+  if (isNaN(parsedSalary) || parsedSalary <= 0) {
+    pushNotification('Salary must be a valid positive number.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateFormData(formData) {
+  return (
+    validateName(formData.name) &&
+    validateAge(formData.age) &&
+    validatePosition(formData.position) &&
+    validateOffice(formData.office) &&
+    validateSalary(formData.salary)
+  );
+}
+
+const form = document.querySelector('.new-employee-form');
+
+document.getElementById('save-button').addEventListener('click', () => {
+  const formData = [...form.elements].reduce((acc, formInput) => {
+    if (formInput.name) {
+      acc[formInput.name] = formInput.value;
+    }
+
+    return acc;
+  }, {});
+
+  if (validateFormData(formData)) {
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${formData.name}</td>
+      <td>${formData.position}</td>
+      <td>${formData.office}</td>
+      <td>${formData.age}</td>
+      <td>$${parseInt(formData.salary).toLocaleString('en-US')}</td>`;
+
+    table.querySelector('tbody').appendChild(newRow);
+    pushNotification('Employee added successfully!', 'success');
+    form.reset();
+  }
+});
+
+// 5. Implement editing of table cells by double-clicking on it. (optional)
+const input = document.createElement('input');
+
+input.className = 'cell-input';
+
+input.addEventListener('blur', () => {
+  const cell = input.parentElement;
+
+  cell.innerText = input.value || input.getAttribute('data-initial');
+});
+
+input.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    const cell = input.parentElement;
+
+    cell.innerText = input.value || input.getAttribute('data-initial');
+  }
+});
+
+document.querySelector('tbody').addEventListener('dblclick', (e) => {
+  const cell = e.target;
+
+  if (cell.tagName.toLowerCase() === 'td') {
+    const initialValue = cell.innerText;
+
+    input.setAttribute('data-initial', initialValue);
+    input.value = initialValue;
+
+    cell.innerText = '';
+    cell.appendChild(input);
+    input.focus();
+  }
+});
```

Comment:
Code quality issue: You are directly manipulating the DOM to edit a cell. Consider using a more declarative approach, for example by storing the edited cell in a state variable and using this to update the DOM.